src/Microsoft.AspNetCore.Authorization.Middleware/AuthorizationEndpointConventionBuilderExtensions.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Routing;
+
+namespace Microsoft.AspNetCore.Builder
+{
+    public static class AuthorizationEndpointConventionBuilderExtensions
+    {
+        // REVIEW: What other overloads should exist?
+        // e.g. params string[] policies, or params string[] roles
+        public static IEndpointConventionBuilder RequireAuthorization(this IEndpointConventionBuilder builder, params IAuthorizeData[] authorizeData)
```

Comment:
I'd suggest starting with `string` (policyname) and a single `IAuthorizeData` here. I don't think passing in multiple datum is that common, and I would expect that a policy name is the most common.

Reply:
So IAuthorizeData already has policy name inside it, so maybe you just need 2 overloads to start one that takes policyName, and another that takes IAuthorizeData

Reply:
I'll setup two overloads:
* `params IAuthorizeData[] authorizeData`
* `params string[] policyNames`

I think param arrays give a good experience for passing in one, and passing in multiple.

Reply:
but why tho? Like why do you want multiple? Why are you **so** in love with complexity?

Reply:
You can have multiple `AuthorizeAttribute`s on a method. It makes sense that the auth helper method would also allow multiple.